src/lib/EDSL.ml

```diff
@@ -111,3 +109,174 @@ let if_seq ~t ?e c =
   match e with
   | None -> if_then c (seq t)
   | Some f -> if_then_else c (seq t) (seq f) 
+
+
+let eprintf fmt l =
+  with_redirections
+    (call (string "printf" :: string "--" :: fmt :: l)) [
+    to_fd (int 1) (int 2);
+  ]
+
+module Command_line = struct
+  type 'a cli_option = {
+    switches: string list;
+    doc: string;
+    default: 'a;
+  }
+  type _ option_spec =
+    | Opt_flag:   bool t cli_option -> bool t option_spec
+    | Opt_string: string t cli_option -> string t option_spec
+  and (_, _) cli_options =
+    | Opt_end: string -> ('a, 'a) cli_options
+    | Opt_cons: 'c option_spec * ('a, 'b) cli_options -> ('c -> 'a, 'b) cli_options
+
+  module Arg = struct
+    let string ?(default = string "") ~doc switches =
+      Opt_string {switches; doc; default}
+    let flag ?(default = bool false) ~doc switches =
+      Opt_flag {switches; doc; default}
+
+    let (&) x y = Opt_cons (x, y)
+    let usage s = Opt_end s
+
+  end
+
+  let parse
+      (options: ('a, unit t) cli_options)
+      (action: anon: string list t -> 'a) : unit t =
```

Comment:
yes doing something like `Cmdliner`'s `pos` thing is left for future work :)

For now we just collect the anonymous arguments and pass them as a list (which is a huge improvement over the previous version that was just dropping them).